### PR TITLE
Expose schemaPath for external schemas

### PR DESCRIPTION
**Pull request by giordanolucas**
Also replaced the logic used to get the schemaPath on ExternalSchemaTypeExpressionNode.java <-- Please review this

### CODE REVIEW

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java [hunk 1]
```diff
@@ -308,6 +310,23 @@ public String internalFragment()
         }
     }
 
+    public String schemaPath()
+    {
+        ResolvedType resolvedType = getResolvedType();
+        if (resolvedType instanceof AbstractExternalType)
+        {
+            if (!((AbstractExternalType) resolvedType).getSchemaPath().equalsIgnoreCase(ARTIFICIAL_NODE))
```
Comment (dfmacias) [inline raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java]: Why did you add this check?

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java [hunk 2]
```diff
@@ -33,6 +34,7 @@
 import javax.annotation.Nullable;
 import javax.json.JsonObject;
 
+import org.apache.commons.lang.StringUtils;
```
Comment (dfmacias) [inline raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java]: unused import

File: raml-parser-2/src/test/java/org/raml/v2/api/XmlExternalFileTestCase.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.raml.v2.api.model.v10.datamodel.TypeDeclaration;
+import org.raml.v2.api.model.v10.datamodel.XMLTypeDeclaration;
+
+import javax.validation.constraints.AssertTrue;
+import java.io.File;
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.*;
```
Comment (dfmacias) [inline raml-parser-2/src/test/java/org/raml/v2/api/XmlExternalFileTestCase.java]: Don't use import * and remove unused ones, please

File: raml-parser-2/src/test/java/org/raml/v2/api/XmlExternalFileTestCase.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    public void checkThatTypeDeclarationRetrievedFromDataTypeFragmentImplementsMostSpecializedTypeDeclarationInterface() throws IOException
+    {
+        File input = new File("src/test/resources/org/raml/v2/api/v10/xml-external/input.raml");
+        assertTrue(input.isFile());
+        RamlModelResult ramlModelResult = new RamlModelBuilder().buildApi(input);
+        assertFalse(ramlModelResult.hasErrors());
+
+        Api apiV10 = ramlModelResult.getApiV10();
+
+        Assert.assertTrue(((XMLTypeDeclaration) apiV10.resources().get(0).methods().get(0).body().get(0)).schemaPath().endsWith("person-schema.xsd"));
```
Comment (dfmacias) [inline raml-parser-2/src/test/java/org/raml/v2/api/XmlExternalFileTestCase.java]: A better approach would be checking if schema path is right compared with raml path `src/test/resources/org/raml/v2/api/v10/xml-external/input.raml`

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java (hunk 1 above)
Comment (giordanolucas) [reply raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java]: No reason, I can remove it if you want.
I thought it was not right to expose that string to the user

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java (hunk 2 above)
Comment (giordanolucas) [reply raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java]: Done, removed

File: raml-parser-2/src/test/java/org/raml/v2/api/XmlExternalFileTestCase.java (hunk 3 above)
Comment (giordanolucas) [reply raml-parser-2/src/test/java/org/raml/v2/api/XmlExternalFileTestCase.java]: Done, removed.
